### PR TITLE
feat: Update API service

### PR DESCRIPTION
**Pull request by michaljurecko**
Jira: https://keboola.atlassian.net/browse/PSGO-596

All tests will pass in the final [PR](https://github.com/keboola/keboola-as-code/pull/1770).

[Large set of fixes and E2e tests](https://github.com/keboola/keboola-as-code/pull/1759) is divided into smaller PRs to do a review.

--------------

**Changes:**
- Updated API service and dependencies, based on changed layers bellow the API layer.

-----------

Commend per commits. Please, explore [the final version](https://github.com/keboola/keboola-as-code/pull/1759) in the IDE.

-------------

### CODE REVIEW

File: api/stream/design.go [hunk 1]
```diff
@@ -257,30 +265,17 @@ var _ = Service("stream", func() {
...
 			SourceNotFoundError()
 			ForbiddenProtectedSettingError()
 		})
 	})
 
-	Method("RefreshSourceTokens", func() {
-		Meta("openapi:summary", "Refresh source tokens")
-		Description("Each sink uses its own token scoped to the target bucket, this endpoint refreshes all of those tokens.")
-		Result(Source)
-		Payload(GetSourceRequest)
```
Comment (michaljurecko) [inline api/stream/design.go:275]: Unused, token can be refreshed by disable/enable API calls (not implemented yet).

File: internal/pkg/service/stream/api/mapper/sink_request.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+		// Template column
+		if tmplColumn, ok := columnEntity.(column.Template); ok {
+			if columnPayload.Template == nil {
+				return table.Mapping{}, svcerrors.NewBadRequestError(errors.Errorf(`column "%s" is missing template`, columnPayload.Name))
+			}
+
+			// if err := m.jsonnetValidator.Validate(columnPayload.Template.Content); err != nil {
+			//	return table.Mapping{}, svcerrors.NewBadRequestError(errors.Errorf(`column "%s" template is invalid: %w`, columnPayload.Name, err))
+			//}
```
Comment (hosekpeter) [inline internal/pkg/service/stream/api/mapper/sink_request.go:139]: remove that

**Review by hosekpeter — APPROVED** [review]

File: internal/pkg/service/stream/api/service/sink.go [hunk 3]
```diff
@@ -2,53 +2,220 @@ package service
...
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/dependencies"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/model"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
 )
 
-func (s *service) CreateSink(_ context.Context, d dependencies.SourceRequestScope, payload *stream.CreateSinkPayload) (res *stream.Task, err error) {
-	return nil, errors.NewNotImplementedError()
-}
+//nolint:dupl // CreateSource method is similar
+func (s *service) CreateSink(ctx context.Context, d dependencies.SourceRequestScope, payload *api.CreateSinkPayload) (*api.Task, error) {
```
Comment (Matovidlo) [inline internal/pkg/service/stream/api/service/sink.go:24]: `d dependencies.SinkRequestScope`?  I would expect that the source dependencies are for `CreateSource` function.

File: internal/pkg/service/stream/api/service/sink.go [hunk 4]
```diff
@@ -2,53 +2,220 @@ package service
...
+	if err != nil {
+		return nil, err
+	}
+
+	return s.mapper.NewSinkResponse(sink)
 }
 
-func (s *service) UpdateSink(context.Context, dependencies.SinkRequestScope, *stream.UpdateSinkPayload) (res *stream.Task, err error) {
-	return nil, errors.NewNotImplementedError()
+func (s *service) ListSinks(ctx context.Context, d dependencies.SourceRequestScope, payload *api.ListSinksPayload) (*api.SinksList, error) {
```
Comment (Matovidlo) [inline internal/pkg/service/stream/api/service/sink.go:72]: `d dependencies.SinkRequestScope`?  I would expect that the source dependencies are for `ListSources` function.

File: internal/pkg/service/stream/api/mapper/sink_request.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	}
+
+	// Type specific updates
+	switch entity.Type {
+	case definition.SinkTypeTable:
+		if entity.Table == nil {
+			entity.Table = &definition.TableSink{}
+		}
+		if payload.Table != nil {
+			if err := m.updateTableSinkEntity(entity.Table, payload); err != nil {
```
Comment (Matovidlo) [inline internal/pkg/service/stream/api/mapper/sink_request.go:73]: Cannot be check of  `payload.Table` inside `updateTableSinkEntity`? The `entity.Table` is set before but we do not push just the `payload.Table` but whole `payload`. So we could do the check in the function where we will return `nil` error and reduce nest of code.

File: internal/pkg/service/stream/api/mapper/source_request.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+	// Type specific updates
+	switch entity.Type {
+	case definition.SourceTypeHTTP:
+		if entity.HTTP == nil {
+			entity.HTTP = &definition.HTTPSource{}
+		}
+		if entity.HTTP.Secret == "" {
+			entity.HTTP.Secret = idgenerator.StreamHTTPSourceSecret()
+		}
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply internal/pkg/service/stream/api/mapper/source_request.go:70]: @jachym-tousek-keboola do we need some additional comment?

**Pull request merged**